server/client.go

```diff
@@ -709,6 +725,7 @@ func (c *client) deliverMsg(sub *subscription, mh, msg []byte) {
 
 	deadlineSet := false
 	if client.bw.Available() < (len(mh) + len(msg) + len(CR_LF)) {
+		atomic.AddInt64(&c.cache.wfc, 1)
```

Comment:
Ok, I think I got it. During the delivery process, we mark the clients that had a too small buffer. Then, in the readLoop, for those clients that need to be flushed, we check if we were notified. If I understand correctly, why do we lookup in the for loop and not before entering the loop?